### DISCUSSION TITLE
Publishing the extension on Open VSX

### DISCUSSION THREAD

**Discussion by CTC-Portofolio** [cyrilletuzi/vscode-angular-schematics#666] [Ideas]
Hello,

Cloud Workspaces are getting more and more common, and except GitHub's own Codespaces, I don't know any other provider that uses the default VS Code - the one which has access to the Microsoft`s Marketplace. 

As the other providers use a custom VS Code which has access only to Open VSX, I kindly ask if you could publish this awesome extension the Open VSX registry. It could help a lot ^_^.

Please refer to this [gitpod post](https://www.gitpod.io/blog/open-vsx) for more context and explanations!

Have a good day!

**Comment by cyrilletuzi** [comment]
Hello,

I wasn’t aware of these alternatives tools. I will take a deeper look next week (can’t this week), but at first glance I don’t see any reason to be opposed to this.

On the technical side, is there any testing to do, or are we sure that an extension working in VS Code will work in alternative tools (ie. the VS Code extension API is fully available in the alternative tools)?

Note that for a cloud use, more work may be needed. I’ve already started to do it in the v5 branch. As I’m not using cloud workspaces myself, I would be glad if someone can provide a test setup to check everything is working in this scenario.

**Comment by CTC-Portofolio** [comment]
Thank you for your reply. I can help with that, by doing some basic testing as such:

1. Download the VSIX from the Microsoft Store

2. Install it by hand on the Gitpod VsCode Cloud instance

3. Create new components/directives/modules/etc with it.

I'll post my results here. 

P.S: The cloud workspaces are Ephemeral, and that means they get reset everytime you close them. That means I have to reinstall using the VSIX every single time, that's why I made this request 😁

**Comment by CTC-Portofolio** [comment]
I tested it, and it works out of the box, no other config/workaround/whatever needed. 
I think it`s ok to publish it on Open VSX too as it is.  
Please refer [here](https://www.gitpod.io/blog/open-vsx#publishing) on how to do it.

**Comment by cyrilletuzi** [comment]
Done: [https://open-vsx.org/extension/cyrilletuzi/angular-schematics](https://open-vsx.org/extension/cyrilletuzi/angular-schematics)

(The warning should be gone soon, it requires manual validation from the Eclipse team: EclipseFdn/open-vsx.org#637)

I've automated the process, so it will get updates too.

Thanks for bringing this to my attention.

Please consider:
- adding 5 stars on [VS Code Marketplace](https://marketplace.visualstudio.com/items?itemName=cyrilletuzi.angular-schematics)
- adding 5 stars on [OpenVSIX](https://open-vsx.org/extension/cyrilletuzi/angular-schematics)
- becoming a [sponsor](https://github.com/sponsors/cyrilletuzi) or doing a [one-time donation](https://github.com/sponsors/cyrilletuzi?frequency=one-time)

**Comment by CTC-Portofolio** [comment]
Thank you very much for your effort ^_^
